tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,140 @@
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Starting test number 2 - agent should remain offline.")
+
+    start_time = time.time()
+    # extend the test timeout here a bit since we are wrapping bluechi-is-online
+    with Timeout(
+        WAIT_PARAM_VALUE + 1, "bluechi-is-online did not exit within the specified time"
+    ):
+        result = node_foo.bluechi_is_online.agent_is_online(WAIT_PARAM_VALUE)
+
+    assert (
+        not result
+    ), f"Expected bluechi-is-online with --wait={WAIT_PARAM_VALUE} on node {NODE_FOO} to exit false"
+    assert (
+        start_time - time.time() <= WAIT_PARAM_VALUE + 1
+    ), "Expected around 5 second for bluechi-is-online to exit"
+
+    # Test 3: Stop agent, run 'bluechi-is-online agent --wait', start agent and verify --wait returns 0 before the
+    # wait time expires
+    try:
+        with Timeout(MAX_WAIT_PARAM_VALUE, "Timeout during Test 3"):
+            LOGGER.debug(
+                "Starting test number 3, stopping agent to ensure it is inactive before starting `bluechi-is-online`."
+            )
+            node_foo.systemctl.stop_unit("bluechi-agent")
+            assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+            LOGGER.debug("Agent confirmed inactive after stopping.")
+
+            result_future_wait = ResultFuture()
+            start_time = time.time()
+            LOGGER.debug(
+                "Starting `bluechi-is-online` thread with wait time of 20 seconds."
+            )
+            check_thread_wait = threading.Thread(
+                target=check_agent,
+                args=(
+                    node_foo.bluechi_is_online,
+                    result_future_wait,
+                    "Checking if agent comes online",
+                    MAX_WAIT_PARAM_VALUE,
+                    0,
+                ),
+            )
+            check_thread_wait.start()
+            time.sleep(SLEEP_DURATION)
+            node_foo.systemctl.start_unit("bluechi-agent")
+            assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "active")
+            LOGGER.debug("Agent confirmed active after starting.")
+
+            # Wait for the thread to complete
+            with Timeout(2, "Thread didn't stop within 2s of starting bluechi-agent"):
+                check_thread_wait.join()
+
+            elapsed_time = time.time() - start_time
+            LOGGER.debug(
+                f"Test 3 result: {result_future_wait.result}, Elapsed time: {elapsed_time:.2f} seconds"
+            )
+            assert (
+                result_future_wait.result
+            ), "Expected agent to come online before wait expired"
+            assert (
+                2 < elapsed_time < 3
+            ), f"Command took too long to complete: {elapsed_time:.2f} seconds"
+    except TimeoutError as ex:
```

Comment:
I'd remove try block, if Timeout block raised exception, we want the test to fail if Timeout block raises exception